faros_event.sh

```diff
@@ -402,432 +345,280 @@ function resolveDefaults() {
+       ! [ -z "$artifact_org" ] &&
+       ! [ -z "$artifact_repo" ] &&
+       ! [ -z "$artifact_id" ]; then
+        request_body=$(jq \
+            --arg artifact_source "$artifact_source" \
+            --arg artifact_org "$artifact_org" \
+            --arg artifact_repo "$artifact_repo" \
+            --arg artifact_id "$artifact_id" \
+            '.data.artifact +=
+            {
+                "source": $artifact_source,
+                "organization": $artifact_org,
+                "repository": $artifact_repo,
+                "id": $artifact_id
 
-function addCICDObjectsToEvent() {
-    makeOrganization
-    makePipeline
-    request_body=$(jq ".entries += [
-        $cicd_Organization,
-        $cicd_Pipeline
-    ]" <<< $request_body)
+            }' <<< $request_body
+        )
+    fi
 }
 
-function addCICDRepositoryToEvent() {
-    makeArtifactRepository
-    request_body=$(jq ".entries += [
-        $cicd_Repository
-    ]" <<< $request_body)
+function addRunToData() {
+    if ! [ -z "$run_source" ] &&
+       ! [ -z "$run_org" ] &&
+       ! [ -z "$run_pipeline" ] &&
+       ! [ -z "$run_id" ]; then
+        request_body=$(jq \
+            --arg run_source "$run_source" \
+            --arg run_org "$run_org" \
+            --arg run_pipeline "$run_pipeline" \
+            --arg run_id "$run_id" \
+            '.data.run +=
+            {
+                "source": $run_source,
+                "organization": $run_org,
+                "pipeline": $run_pipeline,
+                "id": $run_id
+            }' <<< $request_body
+        )
+    fi
+    if ! [ -z "$run_status" ]; then
+        request_body=$(jq \
+            --arg run_status "$run_status" \
+            '.data.run +=
+            {
+                "status": $run_status
+            }' <<< $request_body
+        )
+    fi
+    if ! [ -z "$run_status_details" ]; then
+        request_body=$(jq \
+            --arg run_status_details "$run_status_details" \
+            '.data.run +=
+            {
+                "statusDetails": $run_status_details,
+            }' <<< $request_body
+        )
+    fi
+    if ! [ -z "$run_start_time" ]; then
+        request_body=$(jq \
+            --arg run_start_time "$run_start_time" \
+            '.data.run +=
+            {
+                "startTime": $run_start_time|tonumber
+            }' <<< $request_body
+        )
+    fi
+    if ! [ -z "$run_end_time" ]; then
+        request_body=$(jq \
+            --arg run_end_time "$run_end_time" \
+            '.data.run +=
+            {
+                "endTime": $run_end_time|tonumber
+            }' <<< $request_body
+        )
+    fi
 }
 
 function sendEventToFaros() {
     log "Sending event to Faros..."
 
     http_response=$(curl --retry 5 --retry-delay 5 \
         --silent --write-out "HTTPSTATUS:%{http_code}" -X POST \
-        $url/graphs/$graph/revisions \
+        "$url/graphs/$graph/events?validateOnly=$validate_only" \
```

Comment:
so it's ok to pass `validate_only=false`?

Reply:
`validateOnly=false` is fine to pass to the API. The `--validate_only` flag does not take any input and is `true` if present `false` otherwise.